CDCgov/prime-reportstream
# add index for item-scoped error message for invalid message type

This PR ensures the correct error message is returned for incorrect HL7 message types when submitted to the full ELR pipeline.

Test Steps:
1. Submit an HL7 message with a type other than OR_R01 to the full ELR pipeline
2. Verify the error message returned specifies that the type submitted is not supported.

## Changes
- Update item logger to include index

## Checklist

### Testing
